### ISSUE TITLE
Making note invisible will not hide the ledger line.

### ISSUE THREAD

**Issue by Attackoh** [musescore/MuseScore#16665]
### Issue type

Engraving bug

### Bug description

Making the top note of a chord invisible should hide unnecessary ledger lines and elongated stems. The ledger line should be automatically hidden or their should be an option for the user to manually hide them.

### Steps to reproduce

1. Make chord above staff with ledger lines
2. Make notes invisible
3. Ledger lines remain

### Screenshots/Screen recordings

![image](https://user-images.githubusercontent.com/120669196/222968173-4a97b677-9ec8-432a-896d-ee75f8003134.png)
![image](https://user-images.githubusercontent.com/120669196/222968197-0063b577-1e75-4900-88e3-6038f1181a8b.png)

### MuseScore Version

4.0.1-230121751

### Regression

No.

### Operating system

OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.0.1-230121751, revision: github-musescore-musescore-9b70a8c

**Comment by Jojo-Schmitz** [comment]
As far as ledger lines are concerned this is a regression vs. Mu3. But not as far as theelongated stem
![image](https://user-images.githubusercontent.com/1786669/222973471-f3e9a6b1-81db-4335-8caf-5ae7ddc9cb7b.png)
![image](https://user-images.githubusercontent.com/1786669/222973503-47bc8b8c-a957-465b-8e20-ce43a03eecc4.png)

**Comment by MarcSabatella** [comment]
In 4.0.2 builds, hiding the notehead hides the ledger lines too.

If you check previous forum and issue tracker threads, you'll there is a long history of people pointing out different use cases that have different needs, and different opinions on which ones should take priority.  For some use cases, it absolutely makes sense to keep the ledger lines, for others, it absolutely does not.  There won't be an answer that satisfies them all.  Ideally, there should be some sort of control for this, so whichever the default is, you can override it.

**Comment by oktophonie** [comment]
I struggled to reproduce this (in any version) until I realised what this problem is more specifically: if you're hiding a top note of the chord above the stave, the ledger lines will not be hidden if there is also any note with ledger lines _below_ the stave; even more confusingly, it works 'as expected' vice versa:
![image](https://user-images.githubusercontent.com/20604032/223083986-27307d30-7c35-4742-9c90-3ddfde5ca793.png)

This is identical in 3.6. I will close this as (essentially) a duplicate of https://github.com/musescore/MuseScore/issues/16665 but will reference it from there. I quite agree that per-chord options are needed for how to handle the visibility of ledger lines and stems extending to invisible notes.
